template/.github/workflows/ios_deploy_staging_to_firebase.yml

```
@@ -0,0 +1,52 @@
+name: ios-deploy-to-staging
+on:
+  # Trigger the workflow on push action
+  push:
+    branches:
+      - develop
+
+jobs:
+  build_and_upload_staging_app_to_firebase:
+    name: Build And Upload iOS Staging Application To Firebase
+    runs-on: macOS-latest
+    env:
+      TEAM_ID: ${{ secrets.TEAM_ID }}
+      FASTLANE_USER: ${{ secrets.FASTLANE_USER }}
+      FASTLANE_PASSWORD: ${{ secrets.FASTLANE_PASSWORD }}
+      FASTLANE_APPLE_APPLICATION_SPECIFIC_PASSWORD: ${{ secrets.FASTLANE_APPLE_APPLICATION_SPECIFIC_PASSWORD }}
+      FASTLANE_SESSION: ${{ secrets.FASTLANE_SESSION }}
+      MATCH_PASSWORD: ${{ secrets.MATCH_PASSWORD }}
+      KEYCHAIN_PASSWORD: ${{ secrets.KEYCHAIN_PASSWORD }}
+    steps:
+    - uses: actions/checkout@v1
+
+    - name: Install SSH key
+      uses: webfactory/ssh-agent@v0.4.1
+      with:
+        ssh-private-key: ${{ secrets.SSH_PRIVATE_KEY }}
+
+    - uses: subosito/flutter-action@v1
+      with:
+        channel: 'stable'
+        flutter-version: '3.0.4'
+
+    - name: Get flutter dependencies
+      run: flutter pub get
+
+    - name: Run code generator
+      run: flutter packages pub run build_runner build --delete-conflicting-outputs
+
```

There is 1 missing step to inject the env credentials before building

```suggestion
    - name: Set up .env.staging
      env:
        ENV_STAGING: ${{ secrets.ENV_STAGING }}
      run: |
        echo $ENV_STAGING > .env.staging
```